### PR TITLE
Drop the cruft

### PR DESCRIPTION
**Pull request by cstamas**
Continued along the lines of https://github.com/apache/maven-gpg-plugin/pull/65

Drop even more cruft
Changes:
* drop MAT
* simplify all the hoops and loops
* stop propagating bad practice (passphrase in files or SCM)
* introduce another signer: BC signer

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java [hunk 1]
```diff
@@ -177,7 +188,10 @@ AbstractGpgSigner newSigner(MavenProject project) throws MojoExecutionException,
         signer.setLockMode(lockMode);
         signer.setArgs(gpgArguments);
 
-        loadGpgPassphrase();
+        String passphrase = System.getenv(MAVEN_GPG_PASSPHRASE);
```
Comment (gnodet) [inline src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java:191]: Shouldn’t the env variable be configurable with the current value as a default ?

**Comment by hboutemy** [comment]
splitting at least the MAT migration vs the gpg passphrase change would ease review and learning

**Comment by cstamas** [comment]
> splitting at least the MAT migration vs the gpg passphrase change would ease review and learning

Once @olamy or @slawekjaranowski figure out this PR https://github.com/apache/maven-gh-actions-shared/pull/96 (ideally as "v4"), will split this PR into several distinct PRs

File: src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java (hunk 1 above)
Comment (cstamas) [reply src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java]: Fixed

**Comment by cstamas** [comment]
This PR is going to be "chopped" into a series of smaller PRs, along the lines for 3.2.0 version:
https://issues.apache.org/jira/issues/?jql=project%20%3D%20MGPG%20AND%20fixVersion%20%3D%203.2.0

**Pull request closed without merging**